iptables/builder/builder_nat.go

```diff
@@ -246,30 +248,71 @@ func addOutputRules(cfg config.Config, dnsServers []string, nat *table.NatTable)
 	return nil
 }
 
-func buildNatTable(
-	cfg config.Config,
-	dnsServers []string,
-	loopback string,
-	ipv6 bool,
-) (*table.NatTable, error) {
-	prefix := cfg.Redirect.NamePrefix
-	inboundRedirectChainName := cfg.Redirect.Inbound.RedirectChain.GetFullName(prefix)
-	inboundChainName := cfg.Redirect.Inbound.Chain.GetFullName(prefix)
-	nat := table.Nat()
-
+func addPreroutingRules(cfg config.Config, nat *table.NatTable, ipv6 bool) error {
+	inboundChainName := cfg.Redirect.Inbound.Chain.GetFullName(cfg.Redirect.NamePrefix)
 	if cfg.Log.Enabled {
 		nat.Prerouting().Append(
 			Jump(Log(PreroutingLogPrefix, cfg.Log.Level)),
 		)
 	}
+
+	if len(cfg.Redirect.VNet.Networks) > 0 {
+		interfaceAndCidr := map[string]string{}
+		for i := 0; i < len(cfg.Redirect.VNet.Networks); i++ {
+			// we accept only first : so in case of IPv6 there should be no problem with parsing
+			pair := strings.SplitN(cfg.Redirect.VNet.Networks[i], ":", 2)
```

Comment:
Can't we just have a pair in the structure? Why do we need to parse this here? with IPv6 having `:` as a delimiter it's a bit confusing that we have another `:` in front of it.